scripts/qa_crowbarsetup.sh

```diff
@@ -4103,6 +4111,14 @@ function onadmin_prepare_cloudupgrade()
     # recreate the SUSE-Cloud Repo with the latest iso
     onadmin_prepare_cloud_repos
     onadmin_add_cloud_repo
+    # FIXME: ugly hack - we need to prepare also SP1 Cloud repositories
+    # to correctly install SP1 nodes on SP2-enabled Cloud7
+    if iscloudver 7plus && [[ $want_sles12sp2 ]] ; then
+        unset want_sles12sp2
+        onadmin_prepare_cloud_repos
+        onadmin_add_cloud_repo
```

Comment:
to reduce code duplication we could move the extra calls into the prepare and add functions. Like one level of recursion